source/simulator/solver.cc

```diff
@@ -955,12 +1102,24 @@ namespace aspect
           stokes_A_block_is_symmetric(),
           parameters.linear_solver_A_block_tolerance);
         const internal::BlockSchurPreconditioner<internal::InverseVelocityBlock<LinearAlgebra::PreconditionAMG>,
-              internal::InverseWeightedMassMatrix<TrilinosWrappers::PreconditionBase>>
+              internal::SchurComplementOperator>
               preconditioner_expensive (system_matrix,
                                         system_preconditioner_matrix,
-                                        inverse_weighted_mass_matrix,
+                                        *schur,
                                         inverse_velocity_block_expensive);
 
+
+
+
+
+
+        // create an expensive preconditioner that solves for the A block with CG
+
+
+
+
+
+
```

Comment:
I think the number of empty lines here is a bit excessive ;-). Reduce to one or at most two.